cmd/grafana-agent/Dockerfile

```diff
@@ -44,6 +44,15 @@ EOF
 COPY --from=build /src/agent/build/grafana-agent /bin/grafana-agent
 COPY cmd/grafana-agent/agent-local-config.yaml /etc/agent/agent.yaml
 
+ARG UID=0
+ARG GF_UID=473
+ARG GF_UNAME="grafana-agent"
+
+RUN groupadd --gid $GF_UID $GF_UNAME
+RUN useradd -m -u $GF_UID -g $GF_UID $GF_UNAME
+RUN chown -R $GF_UNAME:$GF_UNAME /etc/agent
+RUN chown -R $GF_UNAME:$GF_UNAME /bin/grafana-agent
+USER $UID:$UID
```

Comment:
Do we really need a second Docker container? You can specify a non-root user to run as when running the normal container, can't you?

Reply:
Good point. Some users reported having issues with that, but maybe having a valid user in the container will make that easier?

Reply:
https://stackoverflow.com/a/57917406 gives an example of security context with something like this.